include/dlaf/mc/cholesky.h

```diff
@@ -0,0 +1,309 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/include/parallel_executors.hpp>
+#include <hpx/include/threads.hpp>
+
+#include "dlaf/blas_tile.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/common/pipeline.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/communication/functions_sync.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/util_matrix.h"
+
+/// @file
+
+namespace dlaf {
+
+static bool use_pools = true;
+
+/// @brief Cholesky implementation on local memory
+///
+/// @param uplo specifies that the matrix is \a Lower triangular
+/// @tparam mat refers to a dlaf::Matrix object
+///
+/// @throws std::runtime_error if \p uplo = \a Upper decomposition is chosen (not yet implemented)
+
+template <class T>
+void cholesky(blas::Uplo uplo, Matrix<T, Device::CPU>& mat) {
+  // Check that matrix is on local memory
+  if (mat.distribution().commGridSize() == comm::Size2D{1, 1}) {
```

Comment:
Why this check has not been added as the square size checking?

Reply:
you're right.. I'm gonna add it to `util_matrix.h`